grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/Reasoner.java

```diff
@@ -162,10 +160,10 @@ public static void precomputeInferences(GraknGraph graph){
         }
         Set<VarName> selectVars = inputQuery.admin().getSelectedNames();
         Iterator<Conjunction<VarAdmin>> conjIt = inputQuery.admin().getPattern().getDisjunctiveNormalForm().getPatterns().iterator();
-        ReasonerQueryImpl conjunctiveQuery = new ReasonerMatchQuery(graph.graql().match(conjIt.next()).select(selectVars), graph);
+        ReasonerQueryImpl conjunctiveQuery = new ReasonerQueryImpl(graph.graql().match(conjIt.next()).select(selectVars), graph);
         Stream<Map<VarName, Concept>> answerStream = conjunctiveQuery.resolve(materialise);
         while(conjIt.hasNext()) {
-            conjunctiveQuery = new ReasonerMatchQuery(graph.graql().match(conjIt.next()).select(selectVars), graph);
+            conjunctiveQuery = new ReasonerQueryImpl(graph.graql().match(conjIt.next()).select(selectVars), graph);
```

Comment:
Does this mean any query can be processed? Not only match queries? I ask because if that is not the case then this new name is confusing.